### PR TITLE
LA-398 Prevent duplicate haproxy conf files

### PR DESCRIPTION
**Pull request by evrardjp**
HAProxy configuration files have changed names between early
kilo and newton, and therefore some frontend/backends are defined
too many times.

A cleanup of the conf.d files will ensure we start fresh on
newton during the haproxy play.

The impact is limited because:
- All our services are already down for the maintenance;
- We archive the files but don't reload the load balancer;
- We let haproxy do the needful when need be.

Issue: [LA-398](https://rpc-openstack.atlassian.net/browse/LA-398)

### CODE REVIEW

**Comment by evrardjp** [comment]
Removed WIP.

**Comment by evrardjp** [comment]
Weirdly I ran into an issue on r11.1.18 that I didn't have with r11.0.0. Haproxy installation of the package failed because it tries to restart with the existing /etc/init.d/haproxy service file, which still holds the loading of conf.d files. Under investigation.

**Comment by evrardjp** [comment]
On top of it, re-running the ubuntu14 leap script would fail because the cp fails, because the folder exists.

Back into WIP.

**Comment by evrardjp** [comment]
recheck_cit_trusty_leapfrogupgrade_swift -- tempest issue.

**Comment by evrardjp** [comment]
Rebased.

**Comment by evrardjp** [comment]
recheck_cit_all

**Comment by evrardjp** [comment]
recheck_cit_artifact

**Comment by Frank-ZhangXin** [comment]
recheck_cit_artifact
recheck_cit_trusty_deploy_swift
recheck_cit_trusty_leapfrogupgrade_swift

File: scripts/leapfrog/pre_redeploy.sh [hunk 1]
```diff
@@ -112,3 +112,11 @@ popd
 pushd ${RPCO_DEFAULT_FOLDER}/rpcd/playbooks
     openstack-ansible configure-apt-sources.yml
 popd
+if [[ ! -f "${UPGRADE_LEAP_MARKER_FOLDER}/cleanup-haproxy.complete" ]]; then
+    # Ad-hoc commands to cleanup haproxy config dir.
+    ansible -m shell -a "(mkdir /root/haproxy-leap/ && cp -r /etc/haproxy/ /root/haproxy-leap/) || true" haproxy -i /opt/rpc-openstack/openstack-ansible/playbooks/inventory
+    ansible -m shell -a "cp /etc/init.d/haproxy /root/haproxy-leap/haproxy.kilo || true" haproxy -i /opt/rpc-openstack/openstack-ansible/playbooks/inventory
+    ansible -m shell -a "rm -f /etc/haproxy/conf.d/* || true" haproxy -i /opt/rpc-openstack/openstack-ansible/playbooks/inventory
+    ansible -m shell -a "cp /etc/init.d/haproxy.dpkg-dist /etc/init.d/haproxy || rm /etc/init.d/haproxy || true" haproxy -i /opt/rpc-openstack/openstack-ansible/playbooks/inventory
```
Comment (Frank-ZhangXin) [inline scripts/leapfrog/pre_redeploy.sh]: Is there any particular reason you didn't separate two commands here?

Comment (evrardjp) [inline scripts/leapfrog/pre_redeploy.sh]: Yes, if you don't have a backup and therefore can't restore, still remove it.
If you re-run, and don't have the file anymore, don't fail.

**Comment by evrardjp** [comment]
We need one more vote with manual testing.

**Comment by evrardjp** [comment]
Don't merge: I got an issue in one of my leapfrog. This could be improved.

**Comment by evrardjp** [comment]
This has to go back to the drawing board. This hack wasn't smart.

**Comment by evrardjp** [comment]
The change is now implemented in OSA OPS.

**Pull request closed without merging**